#!/usr/bin/env python
PACKAGE = "ar_test"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("parent_frame_id",    str_t,    0, "Parent frame id",  "map")
gen.add("child_frame_id",    str_t,    0, "Child frame id",  "frame_0")
gen.add("double_paramr", double_t, 0, "Roll",   0, -3.14,   3.14)
gen.add("double_paramp", double_t, 0, "Pitch",   0, -3.14,   3.14)
gen.add("double_paramy", double_t, 0, "Yaw",   0, -3.14,   3.14)
gen.add("use_q",   bool_t,   0, "Use quat. instead of euler",  True)
gen.add("double_paramqx", double_t, 0, "q:x",   0, -1,   1)
gen.add("double_paramqy", double_t, 0, "q:y",   0, -1,   1)
gen.add("double_paramqz", double_t, 0, "q:z",   0, -1,   1)
gen.add("double_paramqw", double_t, 0, "q:w",   1, -1,   1)

gen.add("x", double_t, 0, "Translation x",   0, -5,   5)
gen.add("y", double_t, 0, "Translation y",   0, -5,   5)
gen.add("z", double_t, 0, "Translation z",   0, -5,   5)

exit(gen.generate(PACKAGE, "ar_test", "CompleteTF"))

